### PR TITLE
Show Completed Date for Documents

### LINKED ISSUE #50: [DOC-217] Show Completed Date for Documents
- The document entity in the database should have a completedDate field
- The field is null per default
- When the status is set to completed, the field is filled with the current date
- The completion date is shown in the recipients view and documents view

<sub>From [SyncLinear.com](https://synclinear.com) | [DOC-217](https://linear.app/documenso/issue/DOC-217/show-completed-date-for-documents)</sub>

### PR DESCRIPTION
**Pull request by dephraiim**
- [x] The document entity in the database should have a completedDate field
- [x] The field is null per default
- [x] When the status is set to completed, the field is filled with the current date
- [x] The completion date is shown in the recipient view and documents view

Closes #50

### CODE REVIEW

**Comment by ElTimuro** [comment]
1. Empty completed field (for whatever reason) should not be shown
image

2. Default Viewstate should not scroll. Maybe Limit Title length (adding ...) and all recipients vertically stacked
image
=> maybe even better as a tooltip on Status i.e.?

File: apps/web/pages/documents/[id]/recipients.tsx [hunk 1]
```diff
@@ -82,6 +83,18 @@ const RecipientsPage: NextPageWithLayout = (props: any) => {
             <h2 className="text-2xl font-bold leading-7 text-gray-900 sm:truncate sm:text-3xl sm:tracking-tight">
               {props.document.title}
             </h2>
+            {props.date.completedDate && (
```
Comment (ElTimuro) [inline apps/web/pages/documents/[id]/recipients.tsx:86]: This needs to be document

File: apps/web/pages/_document.jsx [hunk 2]
```diff
@@ -8,6 +8,11 @@ export default function Document(props) {
     <Html className="h-full scroll-smooth bg-gray-100 font-normal antialiased" lang="en">
       <Head>
         <meta name="color-scheme"></meta>
+        <link rel="preconnect" href="https://fonts.googleapis.com" />
```
Comment (ElTimuro) [inline apps/web/pages/_document.jsx:11]: - The fonts should be loaded from the included files, please remove external resource reference
- If this fixes sth. please create a new issue

**Review by ElTimuro — COMMENTED** [review]
Two small points, otherwise greate

File: apps/web/pages/documents/[id]/recipients.tsx (hunk 1 above)
Comment (dephraiim) [reply apps/web/pages/documents/[id]/recipients.tsx:86]: I'll push a commit fixing it

**Comment by dephraiim** [comment]
I'll push an update.

File: apps/web/pages/_document.jsx (hunk 2 above)
Comment (dephraiim) [reply apps/web/pages/_document.jsx:11]: For the fonts, I wanted to set the font with of the completed date to medium weight but the current font available doesn't have medium weight.

It's either 'bold' or 'regular'

Comment (ElTimuro) [reply apps/web/pages/_document.jsx:11]: Thats by design to keep it clean for now

**Comment by Mythie** [comment]
Hey there, what's the status on this? I agree that this is better displayed as a tooltip on the status column when a document is completed.

**Comment by dephraiim** [comment]
I pushed a new commit removing the status column.

It's only a tooltip now

**Comment by dephraiim** [comment]
@ElTimuro I have fixed the conflicts. You can take a look at it

**Comment by dephraiim** [comment]
@Mythie Can you take a look at this?

**Comment by dephraiim** [comment]
Closing this PR in favor of the updates in the [feat/refresh](https://github.com/documenso/documenso/tree/feat/refresh) branch.

**Pull request closed without merging**